pkg/cmd/init/init.go

```diff
@@ -13,8 +37,155 @@ func InitCommand(f *factory.Factory) *cobra.Command {
 		Long:          f.Localizer.LocalizeByID("init.cmd.long"),
 		Example:       f.Localizer.LocalizeByID("init.cmd.example"),
 		SilenceErrors: true,
-		Run:           func(cmd *cobra.Command, args []string) {},
+		Run: func(cmd *cobra.Command, args []string) {
+			owner := promptGetInput(promptContent{
+				label:    "GitHub Organization or Username",
+				errorMsg: "Please Provide a username",
+			})
+			repo := promptGetInput(promptContent{
+				label:    "GitHub Repo Name",
+				errorMsg: "Please Provide a repo name",
+			})
+			cli_name := promptGetInput(promptContent{
+				label:    "CLI Name",
+				errorMsg: "Please Provide a cli name",
+			})
+
+			templateContext := TemplateContext{
+				Owner:   owner,
+				Repo:    repo,
+				CliName: cli_name,
+			}
+
+			cloneStarter(f)
+			f.Logger.Infoln(color.Info("updating starter code with names"))
+			if err := renderTemplates(templateContext, f); err != nil {
+				f.Logger.Error(err)
+				os.Exit(1)
+			}
+			f.Logger.Infof(color.Success("Your %s CLI has been initialized in this directory.\n"), templateContext.CliName)
+		},
 	}
 
 	return cmd
 }
+
+// promptGetInput returns a string got by prompting the user
+func promptGetInput(pc promptContent) string {
+
+	// validate function for validating prompts
+	validate := func(input string) error {
+		if len(input) == 0 {
+			return errors.New(pc.errorMsg)
+		}
+		return nil
+	}
+
+	prompt := promptui.Prompt{
+		Label:    pc.label,
+		Validate: validate,
+	}
+
+	result, err := prompt.Run()
+	if err != nil {
+		fmt.Printf("Prompt failed %v\n", err)
+		os.Exit(1)
+	}
+
+	return result
+}
+
+// clone a git repository to a local path using go-git
+func cloneStarter(f *factory.Factory) {
+	path, pathErr := os.Getwd()
+	if pathErr != nil {
+		f.Logger.Error(pathErr)
+		os.Exit(1)
+	}
+
+	_, cloneErr := git.PlainClone(path, false, &git.CloneOptions{
+		URL:      "https://github.com/ankithans/charmil-starter-template",
+		Progress: f.IOStreams.Out,
+	})
+	if cloneErr != nil {
+		f.Logger.Error(cloneErr)
+		os.Exit(1)
+	}
+}
+
+// render templates
+func renderTemplates(templateContext TemplateContext, f *factory.Factory) error {
+	path, pathErr := os.Getwd()
+	if pathErr != nil {
+		f.Logger.Error(pathErr)
+		os.Exit(1)
+	}
+
+	// rename cli name as given by user
+	oldPath := path + "/cmd/cli"
+	newPath := path + "/cmd/" + templateContext.CliName
+	if err := os.Rename(oldPath, newPath); err != nil {
+		return err
+	}
+
+	err := filepath.Walk(path,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			// ignore folders
+			ignoreSlice := []string{".git", ".github", ".chglog", ".goreleaser.yml", "CONTRIBUTING.md", "bin"}
```

Comment:
Why we have this?

Reply:
to ignore these folders, while applying template

Reply:
Yes. I get it but why? Those are useful files that are part of starter

Reply:
I didn't have any templates defined in these folders and some of these were creating some error while applying templates. So I had to do this.
But these will be generated untouched as it is.